### ISSUE TITLE
NFC on smartphone

### ISSUE THREAD

**Issue by Rombersoft** [L1L1/cardpeek#102]
Can you application read data from of smartphone which integrated NFC module. I bound bank card to NFC module in my smartphone and I want to read card data doe to attach my smartphone to NFC reader. If it is possible explain how because I try via EMV and get wrong card number

**Comment by uschindler** [comment]
All is working correctly. The card number read from the smartphone's NFC (e.g., Google Pay or Apple Pay) is a virtual one (it's called tokenized). If you pay with your smartphone on the POS, you will also see the last few digits of the virtual number on the receipt.

**Comment by Rombersoft** [comment]
Yes, all is working correctly. But how can I understand that it was red NFC banking card and it was red smartphones NFC (google/apple pay)???

**Comment by uschindler** [comment]
You can't do that, because the smartphone is emulating a bank card.

Still there is one way to figure out if it is Google or Apple Pay, because the Smartphone announces a feature called "Customer Device Cardholder Verification". The current released version of the EMV script does not show this, but the pull request #98 modifies it to show that information in CardPeek.

**Comment by Rombersoft** [comment]
Ok, thank you. Check this as fast as can

**Comment by Rombersoft** [comment]
Hello. I return to this problem. I've already update emv.lua script but I don't see difference between real banking card and smartphone. Maybe you explain me how to see this difference in logs or in screenshot cardpeek application?
